packages/components/src/date-time/test/date.js

```diff
@@ -1,97 +1,118 @@
 /**
  * External dependencies
  */
-import { shallow } from 'enzyme';
 import moment from 'moment';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import 'react-dates/initialize';
 
 /**
  * Internal dependencies
  */
 import DatePicker from '../date';
 
-const TIMEZONELESS_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
-
 describe( 'DatePicker', () => {
-	it( 'should pass down a moment object for currentDate', () => {
-		const currentDate = '1986-10-18T23:00:00';
-		const wrapper = shallow( <DatePicker currentDate={ currentDate } /> );
-		const date = wrapper.children().props().date;
-		expect( moment.isMoment( date ) ).toBe( true );
-		expect( date.isSame( moment( currentDate ) ) ).toBe( true );
+	it( 'should highlight the current date', () => {
+		render( <DatePicker currentDate="2022-05-02T11:00:00" /> );
+
+		expect(
+			screen.getByLabelText( 'Monday, May 2, 2022' ).classList
+		).toContain( 'CalendarDay__selected' );
```

Comment:
I wonder if we could do better than asserting class names which are implementation details. Does `react-dates` add any accessible properties to the selected date?

Reply:
Yeah, I tried to follow [the advice](https://testing-library.com/docs/queries/about#priority) on using accessible queries as much as I could but unfortunately had to use a class name for this as it's all that `react-dates` gives us.

<img width="696" alt="Screen Shot 2022-05-04 at 09 58 30" src="https://user-images.githubusercontent.com/612155/166604713-2d1b992a-b967-4825-9abc-aadbfad865b5.png">

If you click on a date then that date gets `Selected. ` prepended to the label but for some reason `react-dates` doesn't do this for the initial selection which is what we want to test here.

Let me know if you spot a better approach.

Reply:
(Personally I think `react-dates` needs to be changed to use `aria-selected` or something like that but that this is out of scope for this PR 😀)

Reply:
Yeah agree. It should be fixed in `react-dates`' end. Thanks for investigating!

Reply:
Another option would be to use `jest-dom`'s [`toHaveStyle`](https://github.com/testing-library/jest-dom#tohavestyle), but that's also testing implementation detail. I think we can at least use [`toHaveClass`](https://github.com/testing-library/jest-dom#tohaveclass) though, WDYT? (Assuming `jest-dom` is already enabled?)

Reply:
Didn't know about that one! Neat 👍